src/CbcCInterface.jl

```diff
@@ -136,62 +138,70 @@ function loadProblem(prob::CbcModel,
     mat = convert(SparseMatrixCSC{Float64,Int32},constraint_matrix)
     nrow,ncol = size(mat)
 
-    @cbc_ccall loadProblem Cvoid (Ptr{Cvoid}, Int32, Int32, Ptr{CoinBigIndex},
-        Ptr{Int32}, Ptr{Float64}, Ptr{Float64}, Ptr{Float64},
-        Ptr{Float64}, Ptr{Float64}, Ptr{Float64}) prob.p ncol nrow mat.colptr.-Int32(1) mat.rowval.-Int32(1) mat.nzval vec_or_null(Float64, col_lb, ncol) vec_or_null(Float64, col_ub, ncol) vec_or_null(Float64, obj, ncol) vec_or_null(Float64, row_lb, nrow) vec_or_null(Float64, row_ub, nrow)
+    GC.@preserve mat begin
+        @cbc_ccall loadProblem Cvoid (Ptr{Cvoid}, Int32, Int32, Ptr{CoinBigIndex},
+            Ptr{Int32}, Ptr{Float64}, Ptr{Float64}, Ptr{Float64},
+            Ptr{Float64}, Ptr{Float64}, Ptr{Float64}) prob ncol nrow mat.colptr.-Int32(1) mat.rowval.-Int32(1) mat.nzval vec_or_null(Float64, col_lb, ncol) vec_or_null(Float64, col_ub, ncol) vec_or_null(Float64, obj, ncol) vec_or_null(Float64, row_lb, nrow) vec_or_null(Float64, row_ub, nrow)
+    end
 end
 
 function readMps(prob::CbcModel, filename::String)
     check_problem(prob)
     @assert isascii(filename)
-    @cbc_ccall readMps Cint (Ptr{Cvoid}, Ptr{UInt8}) prob.p filename
+    GC.@preserve filename begin
```

Comment:
Preserving `filename` and not `prob` is more confusing now that they both show up directly as arguments to the `ccall`. Isn't it safe to drop the preserve here?

Reply:
Yes, I'm convinced of this now.